src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1051,42 +1175,79 @@ private void UpdateDatabase(TransactionOperationContext context, string type, Bl
                     UpdateValue(index, items, valueNameLowered, valueName, updatedDatabaseBlittable);
                 }
             }
-            finally 
+            catch (Exception e)
             {
+                exception = e;
+                throw;
+            }
+            finally
+            {
+                LogCommand(type, index, exception, updateCommand?.AdditionalDebugInformation(exception));
                 NotifyDatabaseAboutChanged(context, databaseName, index, type, DatabasesLandlord.ClusterDatabaseChangeType.RecordChanged);
             }
         }
-        private void UpdateDatabaseRecordEtagForBackup(TransactionOperationContext context, string type, BlittableJsonReaderObject cmd, long index, Leader leader, ServerStore serverStore)
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
We assume that `info` is usually off, if you want to aggressively inline this, you can, but then the contents of the `if` need to be in another method, to reduce the amount of code that is inlined

Reply:
should that method need to be marked for inline too?